### PR TITLE
Add CurrencyExchangeRate model and cron worker

### PR DESCRIPTION
**Pull request by kewitz**
Refactor and reconcile Currency constants and library.
Translate currency library to TypeScript.
Add CurrencyExchangeModel and migration.
Add a daily worker to track supported currencies exchange rates.

### CODE REVIEW

**Comment by kewitz** [comment]
Rebased and fixed. Ready to merge.

File: server/models/CurrencyExchangeRate.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
+import { Model } from 'sequelize';
+
+import restoreSequelizeAttributesOnClass from '../lib/restore-sequelize-attributes-on-class';
+
+/**
+ * Sequelize model to represent an ExpenseAttachedFile, linked to the `ExpenseAttachedFiles` table.
```
Comment (Betree) [inline server/models/CurrencyExchangeRate.ts:6]: I always forget them too when copy pasting the boilerplate :yum: 

```suggestion
 * Sequelize model to represent an CurrencyExchangeRate, linked to the `CurrencyExchangeRates` table.
```

**Review by Betree — COMMENTED** [review]
If I understand correctly, `CurrencyExchangeRate` is mostly there for recording, we don't yet read values from there right?

**Comment by Betree** [comment]
Code is much cleaner BTW, nice job moving to async/await!

**Comment by kewitz** [comment]
@Betree exactly. We're basically recording in our services so we can use it in the dashboard (metabase) and possibly other reports.

**Review by Betree — APPROVED** [review]
:100:

**Pull request merged**